src/core/lib/gprpp/memory.h

```diff
@@ -58,33 +56,28 @@ inline T* New(Args&&... args) {
 }
 
 // Alternative to delete, since we cannot use it (for fear of libstdc++)
-// We cannot add a default value for can_be_null, because they are used as
-// as friend template methods where we cannot define a default value.
-// Instead we simply define two variants, one with and one without the boolean
-// argument.
-template <typename T, bool can_be_null>
+template <typename T>
 inline void Delete(T* p) {
-  GPR_DEBUG_ASSERT(can_be_null || p != nullptr);
-  if (can_be_null && p == nullptr) return;
+  if (p == nullptr) return;
   p->~T();
   gpr_free(p);
 }
-template <typename T>
-inline void Delete(T* p) {
-  Delete<T, /*can_be_null=*/true>(p);
-}
 
-template <typename T>
 class DefaultDelete {
  public:
+  template <typename T>
```

Comment:
out of curiosity: is it needed to move the template param to the method?

Reply:
Yes, that's actually the whole point of this change.  Previously, the compiler was not able to automatically convert from `UniquePtr<Subclass>` to `UniquePtr<BaseClass>`, because they each had a different deleter type.  Now they have the same deleter type, so the compiler can handle this.

This is why previously, when an API takes `UniquePtr<BaseClass>` and we want to pass a `UniquePtr<Subclass>`, we had to do `UniquePtr<BaseClass>(New<Subclass>(...))` instead of just `MakeUnique<Subclass>(...)`.  It also comes into play when we want to use `std::move()` to pass an existing `UniquePtr<Subclass>` into such an API.  (The latter case is actually what prompted me to look into this again.)

Reply:
Oh yes, my bad. Sorry :-) 

Thanks for the change!